utils/oominfo/oomparser/oomparser.go

```diff
@@ -0,0 +1,158 @@
+package oomparser
+
+import (
+  "bufio"
+  "fmt"
+  "log"
+  "time"
+  "regexp"
+  "io"
+  "strconv"
+  "os"
+)
+
+// struct to hold a list of OomInstance objects
+type  OomParser struct {
+	oomInstances []*OomInstance
+}
+
+// struct that contains information related to an OOM kill instance
+type OomInstance struct {
+	// process id of the process
+	Pid int
+	// the name of the process that is killed
+	ProcessName string
+	// the time that the process was reported to be killed,
+	// accurate to the minute
+	TimeOfDeath time.Time
+	// the absolute name of the container that OOMed
+	ContainerName string
+}
+
+
+type SliceOomInstance []*OomInstance
+
+func (self SliceOomInstance) String() string {
+	returnstring := ""
+	for i, currentOom :=  range self {
+		returnstring += fmt.Sprintf(
+			"oomKill instance %d. pid: %d, processName: %s, timeOfDeath: %v\n",
+			 i, currentOom.Pid, currentOom.ProcessName, currentOom.TimeOfDeath)
+	}
+	return returnstring
+}
+
+// public function to retrieve list of OomInstances in the oomLog struct
+func (self OomParser) GetOomLog() []*OomInstance {
+	return self.oomInstances
+}
+
+// gets the container name from a line and adds it to the oomInstance.
+func getContainerName(line string, currentOomInstance * OomInstance) error {
+	re := regexp.MustCompile(
```

Comment:
lets pull out the compiled regexes to global vars. That way we only have to compile them once when we start up.